### PR TITLE
CSP-714 My profile Edit of Contact Details

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/SFA.DAS.Aan.SharedUi/Models/MemberProfile/MemberProfileViewModel.cs [hunk 1]
```diff
@@ -48,7 +49,7 @@ public MemberProfileViewModel(MemberProfileDetail memberProfileDetail, IEnumerab
         UserRole = (memberProfileDetail.IsRegionalChair) ? MemberUserType.RegionalChair : memberProfileDetail.UserType;
         JobTitle = MapProfilesAndPreferencesService.GetProfileValue(memberProfileMappingModel.JobTitleProfileId, memberProfileDetail.Profiles);
         Biography = MapProfilesAndPreferencesService.GetProfileValue(memberProfileMappingModel.BiographyProfileId, memberProfileDetail.Profiles);
-        LinkedinUrl = MapProfilesAndPreferencesService.GetProfileValue(memberProfileMappingModel.LinkedinProfileId, memberProfileDetail.Profiles);
+        LinkedinUrl = (memberProfileDetail.Profiles.Any(x => x.ProfileId == memberProfileMappingModel.LinkedinProfileId && !string.IsNullOrEmpty(x.Value))) ? string.Concat(UrlConstant.LinkedinUrl, MapProfilesAndPreferencesService.GetProfileValue(memberProfileMappingModel.LinkedinProfileId, memberProfileDetail.Profiles)) : string.Empty;
```
Comment (MarkFCain) [inline src/SFA.DAS.Aan.SharedUi/Models/MemberProfile/MemberProfileViewModel.cs:52]: this is just too complex to have in the middle of a mapping, can you just shift this to a suitably named private method and reference that here. Call it something like 'GetLinkedInUrl', that should be fine.

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.Aan.SharedUi/Services/StringFormatter.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace SFA.DAS.Aan.SharedUi.Services;
+public static class StringFormatter
+{
+    public static string? TrimValue(string? value)
```
[the remark this answers is not in the document]
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.Aan.SharedUi/Services/StringFormatter.cs]: Removed StringFormatter and used trim function at that place

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        sut.ControllerContext = new ControllerContext { HttpContext = new DefaultHttpContext { User = user } };
+        sut.AddContextWithClaim(ClaimsPrincipalExtensions.ClaimTypes.AanMemberId, memberId.ToString());
+        sut.AddUrlHelperMock()
+    .AddUrlForRoute(SharedRouteNames.YourAmbassadorProfile, YourAmbassadorProfileUrl);
+
+        // Act
+        var result = await sut.GetContactDetailViewModel(cancellationToken);
+
+        // Assert
+        Assert.That(result, Is.InstanceOf(editContactDetailViewModel.GetType()));
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs:201]: better way of doing this... `result.Should().BeOfType<EditContactDetailViewModel>()`

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+        // Act
+        var result = sut.Index(cancellationToken);
+
+        // Assert
+        outerApiMock.Verify(a => a.GetMemberProfile(It.IsAny<Guid>(), It.IsAny<Guid>(), false, cancellationToken), Times.Once);
+    }
+
+    [Test, RecursiveMoqAutoData]
+    public static void Index_ShouldReturnEditContactDetailViewModel(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs:72]: This is perhaps duplicated in test `GetContactDetailViewModel_ShouldReturnEditContactDetailViewModel` ?

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/EditPersonalInformationController.cs [hunk 5]
```diff
@@ -65,8 +65,8 @@ public async Task<IActionResult> Post(SubmitPersonalDetailModel submitPersonalDe
         updateMemberProfileAndPreferencesRequest.updateMemberProfileRequest.MemberPreferences = updatePreferenceModels;
 
         List<UpdateProfileModel> updateProfileModels = new List<UpdateProfileModel>();
-        updateProfileModels.Add(new UpdateProfileModel() { MemberProfileId = ProfileIds.Biography, Value = submitPersonalDetailModel.Biography });
-        updateProfileModels.Add(new UpdateProfileModel() { MemberProfileId = ProfileIds.JobTitle, Value = submitPersonalDetailModel.JobTitle });
+        updateProfileModels.Add(new UpdateProfileModel() { MemberProfileId = ProfileIds.Biography, Value = submitPersonalDetailModel.Biography?.Trim() });
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/EditPersonalInformationController.cs:68]: Can add test for trim here as well for both fields.

**Review by DevenKShah — DISMISSED** [review]
Few minor comments to consider. But well done on test coverage. I would like to see some repeated code in test to be eliminated, not that I am too bothered, but where possible for happy path tests.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs (hunk 4 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs:72]: yes it is duplicated indeed. Removed the test

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditContactDetailControllerTests/EditContactDetailControllerGetTests.cs:201]: Done

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/EditPersonalInformationController.cs (hunk 5 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/EditPersonalInformationController.cs:68]: Tests coverage increased and included the trimming tests as well

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**